src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +27,32 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public int hashCode() {
+        if (width != null && length != null && color != null) {
+            int result = 42;
+            result = result * 32 * width.hashCode();
+            result = result * 32 * length.hashCode();
+            result = result * 32 * color.hashCode();
+            return result;
+        }
+        return 0;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+        if (obj.getClass().equals(this.getClass())) {
+            return ((Rectangle) obj).getColor().equals(this.color)
```

Comment:
nice, but better to cast whole object to Rectangle once, than you won't need to cast each field separately